File: bridge_api.py
import bridge_api_internal as bridge
from types import SimpleNamespace

bridge.InitializeApp()

def get_number_of_devices():
    return bridge.GetNumDevices()

def get_device(device_index):
    num_devices = bridge.GetNumDevices()
    if device_index >= num_devices:
        raise Exception("No display " + str(device_index) + " connected. Devices connected: " + str(num_devices))

    info = SimpleNamespace(
        bridge_core_version = bridge.GetHoloPlayCoreVersion(),
        bridge_service_version = bridge.GetHoloPlayServiceVersion(),
        hdmi_name = bridge.GetDeviceHDMIName(device_index),
        device_type = bridge.GetDeviceType(device_index),
        button0_down = bridge.GetDevicePropertyInt(device_index, "/buttons/0") == 1,
        button1_down = bridge.GetDevicePropertyInt(device_index, "/buttons/1") == 1,
        button2_down = bridge.GetDevicePropertyInt(device_index, "/buttons/2") == 1,
        button3_down = bridge.GetDevicePropertyInt(device_index, "/buttons/3") == 1
    )

    window = SimpleNamespace(
        x = bridge.GetDevicePropertyWinX(device_index),
        y = bridge.GetDevicePropertyWinY(device_index),
        w = bridge.GetDevicePropertyScreenW(device_index),
        h = bridge.GetDevicePropertyScreenH(device_index),
        aspect_ratio = bridge.GetDevicePropertyDisplayAspect(device_index)
    )

    shader = SimpleNamespace(
        lenticular_pitch = bridge.GetDevicePropertyPitch(device_index),
        lenticular_tilt = bridge.GetDevicePropertyTilt(device_index),
        center_offset = bridge.GetDevicePropertyCenter(device_index),
        subpixel_size = bridge.GetDevicePropertySubp(device_index),
        fringe = bridge.GetDevicePropertyFringe(device_index),
        red_index = bridge.GetDevicePropertyRi(device_index),
        blue_index = bridge.GetDevicePropertyBi(device_index),
        should_invert = bridge.GetDevicePropertyInvView(device_index),
        vertex_shader = bridge.GetVertexShaderGLSL(),
        fragment_shader = bridge.GetFragmentShaderGLSL()
    )

    quilt = SimpleNamespace(
        recommended_texture_resolution_w = bridge.GetDevicePropertyQuiltX(device_index),
        recommended_texture_resolution_h = bridge.GetDevicePropertyQuiltY(device_index),
        tiling_dimension_x = bridge.GetDevicePropertyTileX(device_index),
        tiling_dimension_y = bridge.GetDevicePropertyTileY(device_index)
    )

    device = SimpleNamespace(
        info = info,
        window = window,
        shader = shader,
        quilt = quilt
    )
    return device

File: bridge_api.c
#include <stdlib.h>
#include <string.h>
#include <stdio.h>
#include <stdbool.h>
#include "bridge_api.h"
#include "LookingGlassCoreSDK/HoloPlayCore/include/HoloPlayCore.h"
#include "LookingGlassCoreSDK/HoloPlayCore/include/HoloPlayShaders.h"

char *InitializeApp(void) {
	hpc_client_error error = hpc_InitializeApp("HoloPlayInfo.c", hpc_LICENSE_NONCOMMERCIAL);
	if (error) {
		return "Bridge Service not running";
	}
	return NULL;
}

void CloseApp(void) {
	hpc_CloseApp();
}

char *GetHoloPlayCoreVersion(void) {
	static char buf[1024];
	hpc_GetHoloPlayCoreVersion(buf, 1000);
	return buf;
}

char *GetHoloPlayServiceVersion(void) {
	static char buf[1024];
	hpc_GetHoloPlayServiceVersion(buf, 1000);
	return buf;
}

int GetNumDevices(void) {
	return hpc_GetNumDevices();
}

char *GetDeviceHDMIName(int device_index) {
	static char buf[1024];
	hpc_GetDeviceHDMIName(device_index, buf, 1000);
	return buf;
}

char *GetDeviceType(int device_index) {
	static char buf[1024];
	hpc_GetDeviceType(device_index, buf, 1000);
	return buf;
}

int GetDevicePropertyInt(int device_index, char *property_str) {
	return hpc_GetDevicePropertyInt(device_index, property_str);
}

int GetDevicePropertyWinX(int device_index) {
	return hpc_GetDevicePropertyWinX(device_index);
}

int GetDevicePropertyWinY(int device_index) {
	return hpc_GetDevicePropertyWinY(device_index);
}

int GetDevicePropertyScreenW(int device_index) {
	return hpc_GetDevicePropertyScreenW(device_index);
}

int GetDevicePropertyScreenH(int device_index) {
	return hpc_GetDevicePropertyScreenH(device_index);
}

float GetDevicePropertyDisplayAspect(int device_index) {
	return hpc_GetDevicePropertyDisplayAspect(device_index);
}

float GetDevicePropertyPitch(int device_index) {
	return hpc_GetDevicePropertyPitch(device_index);
}

float GetDevicePropertyTilt(int device_index) {
	return hpc_GetDevicePropertyTilt(device_index);
}

float GetDevicePropertyCenter(int device_index) {
	return hpc_GetDevicePropertyCenter(device_index);
}

float GetDevicePropertySubp(int device_index) {
	return hpc_GetDevicePropertySubp(device_index);
}

float GetDevicePropertyFringe(int device_index) {
	return hpc_GetDevicePropertyFringe(device_index);
}

int GetDevicePropertyRi(int device_index) {
	return hpc_GetDevicePropertyRi(device_index);
}

int GetDevicePropertyBi(int device_index) {
	return hpc_GetDevicePropertyBi(device_index);
}

int GetDevicePropertyInvView(int device_index) {
	return hpc_GetDevicePropertyInvView(device_index);
}

int GetDevicePropertyQuiltX(int device_index) {
	return hpc_GetDevicePropertyQuiltX(device_index);
}

int GetDevicePropertyQuiltY(int device_index) {
	return hpc_GetDevicePropertyQuiltY(device_index);
}

int GetDevicePropertyTileX(int device_index) {
	return hpc_GetDevicePropertyTileX(device_index);
}

int GetDevicePropertyTileY(int device_index) {
	return hpc_GetDevicePropertyTileY(device_index);
}

char *GetVertexShaderGLSL(void) {
	return (char*)hpc_LightfieldVertShaderGLSL;
}

char *GetFragmentShaderGLSL(void) {
	return (char*)hpc_LightfieldFragShaderGLSL;
}
